Refactor shell script data pipeline

IsaacChanXi opened the pull request:
1. Enabled data pipeline code be ran as a single python script, instead of multiple python scripts
2. Dockerized data pipeline
3. Added new instructions to readme

IsaacChanXi committed:
add python pipeline for execution

IsaacChanXi committed:
refactor code for easier imports

IsaacChanXi committed:
add refactoring of workflow into docker image

IsaacChanXi committed:
uncomment pipeline code

longwind48 reviewed (changes requested):
There should be many files that are no longer used, given the introduction of your pipeline. 
- Do some housekeeping to remove outdated and useless log files and make it clear in your README, what files to edit and files to NEVER edit (like schemes data files). 
- I think the schemes csv files shouldn't be pushed. just include the google drive link for devs to download if they need

IsaacChanXi committed:
remove unused csv files and run steps shell script

IsaacChanXi committed:
remove error logging to csv files

IsaacChanXi committed:
remove unused csv files

IsaacChanXi commented:
> There should be many files that are no longer used, given the introduction of your pipeline.
> 
> * Do some housekeeping to remove outdated and useless log files and make it clear in your README, what files to edit and files to NEVER edit (like schemes data files).
> * I think the schemes csv files shouldn't be pushed. just include the google drive link for devs to download if they need

I have removed the CSV files. They were mostly used for ad hoc updates and not for regular maintenance. I also removed the log_error_to_csv function, so that we don't need to keep track of another csv file.

yevkim commented:
> > There should be many files that are no longer used, given the introduction of your pipeline.
> > 
> > * Do some housekeeping to remove outdated and useless log files and make it clear in your README, what files to edit and files to NEVER edit (like schemes data files).
> > * I think the schemes csv files shouldn't be pushed. just include the google drive link for devs to download if they need
> 
> I have removed the CSV files. They were mostly used for ad hoc updates and not for regular maintenance. I also removed the log_error_to_csv function, so that we don't need to keep track of another csv file.

I do use them. CSV files are the reference of what is loaded in db. 
log_error_to_csv is the only way the scrapping is logging errors . how else can we see which websites can be scraped?

IsaacChanXi commented:
> @IsaacChanXi @longwind48 FYI These csv files are quite important as they are reference I have for what to load in to. Please keep them as is. also all the testing logic I kept in files.

Sure, I will rever the changes then!

IsaacChanXi committed:
add back log error to csv for Main scrape py

IsaacChanXi committed:
add passing of doc ids to process via cmd line

IsaacChanXi committed:
add last time an id was scraped successfully

IsaacChanXi committed:
add refresh condition if update is more than a month ago
